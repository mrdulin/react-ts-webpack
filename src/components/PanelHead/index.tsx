import * as React from 'react';
import {string} from 'prop-types';

// 使用这种方式，需要写模块定义文件，见types/index.d.ts
import * as style from './style.module.scss';
// console.log(style);

// 如果不用上面import的形式导入style，还可以用下面这种方式
// const style = require('./style.module');
// console.log(style);

interface IPanelHeadProps {
  title: string;
  subTitle?: string;
  icon?: string;
}

const PanelHead: React.SFC<IPanelHeadProps> = (props: IPanelHeadProps): React.ReactElement<IPanelHeadProps> => {

  // const iconClass = `icon-${icon}`;

  return (
    <div className={style.panelHeading}>
      <h3 className={style.panelTitle}>
        {props.title}
        {props.subTitle ? <small className={style.panelSubTitle}>{props.subTitle}</small> : null}
      </h3>
    </div>
  );
};

PanelHead.propTypes = {
  title: string,
  subTitle: string,
  icon: string
};

export default PanelHead;
