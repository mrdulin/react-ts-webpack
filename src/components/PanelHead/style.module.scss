.panel-heading {
  height: 40px;
  line-height: 40px;
  padding: 0 10px;
  color: #222;
  // border-bottom: 1px solid #eee;
  background-color: #fff;
}

.panel-heading-with-icon {
  composes: panel-heading;
  padding-left: 30px;
}

.panel-title {
  margin: 0;
  padding: 0;
  position: relative;
  font-size: 13px;
  font-weight: normal;
}

.panel-subTitle {
  float: right;
  font-size: 12px;
  color: #999;
}
