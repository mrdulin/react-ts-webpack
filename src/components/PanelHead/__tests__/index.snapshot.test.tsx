/**
 * Created by dulin on 2017/7/6.
 */
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import PanelHead from '../';

describe('component - PanelHead test suites', () => {
  
  it('renders correctly', () => {
    const component: renderer.Renderer = renderer.create(
      <PanelHead title="主标题" subTitle="副标题"/>
    );
    
    const tree = component.toJSON();
    // console.log(tree);
    expect(tree).toMatchSnapshot();
  });
  
  
});
