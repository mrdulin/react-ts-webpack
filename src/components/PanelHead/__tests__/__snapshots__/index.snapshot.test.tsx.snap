// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component - PanelHead test suites renders correctly 1`] = `
<div
  className="panelHeading"
>
  <h3
    className="panelTitle"
  >
    主标题
    <small
      className="panelSubTitle"
    >
      副标题
    </small>
  </h3>
</div>
`;
